src/index.ts

```diff
@@ -545,14 +547,61 @@ export function movePiece(source: any, target: any, metadata: any) {
   if (move !== null)
     movePieceAfter(move, fen);
 
-  if(game.role === Role.PLAYING_COMPUTER) // Send move to engine in Play Computer mode
-    playEngine.move(game.history.last());
+  if(game.role === Role.PLAYING_COMPUTER) { // Send move to engine in Play Computer mode
+    getComputerMove();
+  } 
 
   showTab($('#pills-game-tab'));
   // Show 'Analyze' button once any moves have been made on the board
   showAnalyzeButton();
 }
 
+// Get Computer's next move either from the opening book or engine
+async function getComputerMove() {
+  var bookMove = '';
+  if(game.category === 'standard') { // only use opening book with normal chess
+    var fen = game.history.get(game.history.last()).fen;
+    var moveNo = History.getMoveNoFromFEN(fen);
+    // Cool-down function for deviating from the opening book. The chances of staying in book 
+    // decrease with each move 
+    var coolDownParams = [
+      { slope: 0.2, shift: 1.0 }, // Difficulty level 1
+      { slope: 0.2, shift: 1.2 }, // 2
+      { slope: 0.2, shift: 1.4 }, // 3
+      { slope: 0.2, shift: 1.6 }, // 4
+      { slope: 0.2, shift: 1.8 }, // 5
+      { slope: 0.2, shift: 2.0 }, // 6
+      { slope: 0.2, shift: 2.5 }, // 7
+      { slope: 0.2, shift: 3.0 }, // 8
+      { slope: 0.2, shift: 3.5 }, // 9
+      { slope: 0.2, shift: 4.0 }, // 10
+    ];
+    let a = coolDownParams[game.difficulty - 1].slope;
+    let b = coolDownParams[game.difficulty - 1].shift;
+    let x = moveNo;
+    var sigma = 1 / (1 + Math.exp(a*x - b));
+    if(Math.random() < sigma) {
+      // Use book move (if there is one)
+      var bookMoves = await getBookMoves(fen);
+      var totalWeight = bookMoves.reduce((acc, curr) => acc + curr.weight, 0);
+      var probability = 0;
+      var rValue = Math.random();
+      for(let bm of bookMoves) {
+        probability += bm.weight / totalWeight; // polyglot moves are weighted based on number of wins and draws
```

Comment:
nit: It looks like there's a move probability (`bm.probability`) that can be used here directly.

Reply:
That's true, I didn't look at this very closely. In their code, which is based on XBoard code, they are doing this:
```for (const move of moves) {
   move.probability = Math.pow(move.weight / weightSum, weightPower)
      .toFixed(1)
}
```

with their default weightPower = 0.2. This would act to bring low and high probability moves closer together. Perhaps this is a good idea? Note the probabilities then aren't normalised so I would still have to use normalising code similar to what I've got. For example when using weightPower 0.2, the probabilities of the starting moves for white become:

move: d2d4 original prob: 0.4279334413973337 adjusted prob (^0.2): 0.186046511627907
move: e2e4 original prob: 0.34089781335803365 adjusted prob (^0.2): 0.186046511627907
move: g1f3 original prob: 0.1173045750769129 adjusted prob (^0.2): 0.1627906976744186
move: c2c4 original prob: 0.10658639055211883 adjusted prob (^0.2): 0.13953488372093023
move: g2g3 original prob: 0.00499520328161699 adjusted prob (^0.2): 0.06976744186046512
move: b2b3 original prob: 0.001257071024512885 adjusted prob (^0.2): 0.06976744186046512
move: f2f4 original prob: 0.0004962122465182441 adjusted prob (^0.2): 0.04651162790697675
move: b1c3 original prob: 0.00023156571504184722 adjusted prob (^0.2): 0.04651162790697675
move: b2b4 original prob: 0.00019848489860729764 adjusted prob (^0.2): 0.04651162790697675
move: c2c3 original prob: 0.00009924244930364882 adjusted prob (^0.2): 0.04651162790697675

So it reduces the probability of d4 and e4 significiantly and raises up Nf3 and c4 a lot.